metagraph/src/annotation/taxonomy/label_to_taxid.cpp

```diff
@@ -0,0 +1,95 @@
+#include "tax_classifier.hpp"
+
+#include "common/utils/string_utils.hpp"
+
+#include "common/logger.hpp"
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+void TaxonomyBase::assign_label_type(const std::string &label, bool *require_accversion_to_taxid_map) {
+    if (utils::starts_with(label, "gi|")) {
+        // e.g.   >gi|1070643132|ref|NC_031224.1| Arthrobacter phage Mudcat, complete genome
+        label_type = GEN_BANK;
+        *require_accversion_to_taxid_map = true;
+    } else if (utils::starts_with(utils::split_string(label, ":")[1], "taxid|")) {
+        // e.g.   >kraken:taxid|2016032|NC_047834.1 Alteromonas virus vB_AspP-H4/4, complete genome
+        label_type = TAXID;
+        *require_accversion_to_taxid_map = false;
+    } else {
+        logger->error("Can't determine the type of the given label {}. Please make sure that the labels are in a recognized format.", label);
+        std::exit(1);
+    }
+}
+
+bool TaxonomyBase::get_taxid_from_label(const std::string &label, TaxId *taxid) const {
+    if (label_type == TAXID) {
+        *taxid = static_cast<uint32_t>(std::stoull(utils::split_string(label, "|")[1]));
+        return true;
+    } else if (TaxonomyBase::label_type == GEN_BANK) {
+        std::string acc_version = get_accession_version_from_label(label);
+        if (not accversion_to_taxid_map.count(acc_version)) {
+            return false;
+        }
+        *taxid = accversion_to_taxid_map.at(acc_version);
+        return true;
+    }
+
+    logger->error("Run get_taxid_from_label() for unknown label {}.", label);
+    std::exit(1);
```

Comment:
`get_taxid_from_label()` is not exposed in the CLI, so printing this error message to the user will be strange.

Ether print something that is exposed to the CLI or make this an assert (in case this can only happen if there is a bug in the code or the flow)

Reply:
replaced, thanks!